### ISSUE TITLE
An argument named "architectures" is not expected here

### ISSUE THREAD

**Issue by hiago-miguel** [npalm/terraform-aws-gitlab-runner#464]
Hey,

With aws provider version `3.59.0`, and version 4.41.1 of this module, I got the following error:

```» terraform validate                                                                                                                                                                  ╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/runner/modules/terminate-instances/main.tf line 20, in resource "aws_lambda_function" "terminate_runner_instances":
│   20:   architectures    = ["x86_64"]
│ 
│ An argument named "architectures" is not expected here.
╵
--------------------------------------
```

The argument `architectures` in the aws_lambda_function resource was released in version 3.61.0:
- https://github.com/hashicorp/terraform-provider-aws/releases/tag/v3.61.0

Since the submodule `terminate-instances` uses this argument, 

```
resource "aws_lambda_function" "terminate_runner_instances" {
  architectures    = ["x86_64"]
  description      = "Lifecycle hook for terminating GitLab runner instances"
...
```
The module should require aws provider ~> 3.61.

Or maybe, it could just remove the `architecures` argument:

> [architectures](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#architectures) - (Optional) Instruction set architecture for your Lambda function. Valid values are ["x86_64"] and ["arm64"]. Default is ["x86_64"]. Removing this attribute, function's architecture stay the same.

---

After updating my provider version to 3.61.0 the issue was fixed.

**Comment by npalm** [comment]
Thanks, will check it

**Comment by npalm** [comment]
Justed tested the module with the 4.0 provder (#460) with the default example. Not able to reproduce the rror you mention.

**Comment by hiago-miguel** [comment]
Yes, the issue only happens when using aws provider < 3.61.
And I have opened this issue because the module's minimum requirement for the aws provider is 3.35.
- https://github.com/npalm/terraform-aws-gitlab-runner/tree/4.41.1#requirements

**Comment by cmavromichalis** [comment]
I also ran into this using main branch, tag 4.41.1 (https://github.com/npalm/terraform-aws-gitlab-runner/releases/tag/4.41.1). Bumping to hashicorp/aws 3.75.1 fixed it.

**Comment by npalm** [comment]
Merged solution #468
